app/controllers/teams_controller.rb

```diff
@@ -35,9 +35,13 @@ def update
   end
 
   def destroy
-    return unless @team.destroy
-    flash[:success] = 'Success'
-    redirect_to teams_path
+    if @team.service_providers.empty? && @team.destroy
+      flash[:success] = 'Success'
+      redirect_to teams_path and return
+    end
+
+    flash[:warning] = I18n.t('notices.team_delete_failed')
+    redirect_back(fallback_location: teams_path) and return
```

Comment:
Do you need the `and return` on this one since it is the last line in the method?

Reply:
probably not!